src/components/GameDetail/GameProgress.vue

```diff
@@ -0,0 +1,423 @@
+<template lang="html">
+  <div :class="['game-progresses', { small }]">
+    <div v-if="hasProgress && !showProgressField" class="progress">
+      <div
+        class="progress-data"
+        v-if="!pie"
+      >
+        <progress
+          class="progress-bar"
+          max="100"
+          :value="localProgress.number"
+        >
+          <div class="progress-bar-fallback">
+            <span :style="style">
+              Progress: {{localProgress.number}}%
+            </span>
+          </div>
+        </progress>
+        <div
+          v-if="localProgress.number"
+          class="progress-bar-label"
+        >
+          {{localProgress.number}}%
+        </div>
+      </div>
+
+      <div
+        v-else
+        class="progress-pie"
+        :data-value="localProgress.number"
+      >
+      </div>
+
+      <button
+        v-if="!small && !pie"
+        class="primary"
+        @click="editProgress"
+      >
+        Edit progress
+      </button>
+    </div>
+
+    <div
+      class="progress-field"
+      v-if="showProgressField && !small && !pie"
+    >
+      <div class="progress-data">
+        <div
+          v-if="localProgress.number"
+          class="progress-bar-label"
+        >
+          {{localProgress.number}}%
+        </div>
+
+        <input
+          class="progress-range"
+          v-model.number="localProgress.number"
+          placeholder="50"
+          type="range"
+          min="0"
+          max="100"
+          step="5"
+          autofocus
+        >
+      </div>
+
+      <div class="progress-action">
+        <div class="progress-edit">
+          <button class="secondary" @click="reset">
+            {{ $t('global.cancel') }}
+          </button>
+
+          <button class="primary" @click="saveProgress">
+            {{ $t('global.save') }}
+          </button>
+        </div>
+
+        <button class="danger" @click="deleteProgress">
+          Delete progress
+        </button>
+      </div>
+    </div>
+
+    <button
+      v-if="!hasProgress &&
+            !showProgressField &&
+            !small &&
+            !pie"
```

Comment:
I would put this logic in a computer property and call it something like `showAddProgressButton` or shorter